### DISCUSSION TITLE
How is the "!" preview function of ragflow implemented? I can't find the corresponding code.

### DISCUSSION THREAD

**Discussion by DAAworld** [infiniflow/ragflow#2741] [Q&A]
![image](https://github.com/user-attachments/assets/06439251-ceb5-40ac-8e9f-bc009cba6d5f)
When answering questions, an “!” will be displayed on the message, and a preview of the source of the quote will appear above it. How is this function implemented?

**Comment by KevinHuSh** [comment]
![image](https://github.com/user-attachments/assets/816dcd7b-2a68-4d9f-b5f1-dfee2394ad4e)
